rules/validator.go

```diff
@@ -178,6 +189,25 @@ func (v *validator) validateRollupTags(ruleName string, tags []string) error {
 			return fmt.Errorf("rollup rule %s does not have required rollup tag: %s, provided rollup tags are %v", ruleName, requiredTag, tags)
 		}
 	}
+
+	return nil
+}
+
+func validateChars(str string, invalidChars []rune) error {
+	if len(invalidChars) == 0 {
+		return nil
+	}
+
+	// Validate that given string doesn't contain an invalid character
+	chars := make(map[rune]struct{}, len(invalidChars))
```

Comment:
It might actually be faster if we simply iterate over the slice.

If not, I'd actually just construct the map once and store it in the validator and reuse it here to avoid reconstructing it on every validation call.

Reply:
Yeah, makes sense - I'll construct the mapping in the validator and just past that along here.